assimilation_code/modules/io/io_filenames_mod.f90

```diff
@@ -665,21 +687,25 @@ end subroutine check_attributes
 !> check integer values are the same
 
 
-subroutine check_attribute_value_int(ncFile, filename, ncVarID, att_string, spvalINT)
+subroutine check_attribute_value_int(ncFile, filename, ncVarID, att_string, att_valINT)
 
 integer,          intent(in) :: ncFile
 character(len=*), intent(in) :: filename
 integer,          intent(in) :: ncVarID
 character(len=*), intent(in) :: att_string
-integer,          intent(in) :: spvalINT
+integer,          intent(in) :: att_valINT
+
+integer :: ret_att_valINT
 
-integer :: ret_spvalINT
+!>@todo FIXME : put the variable name in the error message
 
-if ( nf90_get_att(ncFile, ncVarID, att_string, ret_spvalINT) == NF90_NOERR ) then
-   if (spvalINT /= ret_spvalINT) then
-      write(msgstring,*) ' variable attribute, ', trim(att_string), ' in state', spvalINT, &
-                         ' does not match ', trim(att_string), ' ', ret_spvalINT, ' in ', trim(filename)
-      call error_handler(E_ERR, 'check_attributes', msgstring, source)
+if ( nf90_get_att(ncFile, ncVarID, att_string, ret_att_valINT) == NF90_NOERR ) then
+   if (att_valINT /= ret_att_valINT) then
+      write(string1,*) ' variable attribute "'//trim(att_string)//'" in state is ', att_valINT
+      write(string2,*) ' does not match "'//trim(att_string)//'"', ret_att_valINT
+      write(string3,*) ' in "'//trim(filename)//'"'
+      call error_handler(E_ERR, 'check_attributes', string1, &
+                         source, text2=string2, text3=string3)
```

Comment:
I like putting double-quotes around strings that - when things go wrong - can frequently be empty.